src/main/scala/uk/gov/nationalarchives/dp/client/DataProcessor.scala

```diff
@@ -49,6 +49,22 @@ class DataProcessor[F[_]]()(implicit me: MonadError[F, Throwable]) {
       .getOrElse(me.raiseError(PreservicaClientException(s"Entity type '$entityType' not found for id $entityRef")))
   }
 
+  /** Retrieves the [[EntityClient.EntityType]] Node from an entity response
+    * @param entityRef
+    *   The reference of the entity
+    * @param entityResponse
+    *   The XML response from Preservica
+    * @param entityType
+    *   The [[EntityClient.EntityType]] of the entity.
+    * @return
+    *   An [[scala.xml.Node]] found in the entity response (by searching for [[EntityClient.EntityType]]) wrapped in the
+    *   F effect
+    */
+  def getEntityXml(entityRef: UUID, entityResponse: Elem, entityType: EntityType): F[Node] =
```

Comment:
Is this method name accurate. Maybe `getEntityForType` or getEntityForEntityType`

Reply:
With this method, I wanted to make it clear that it was different from `getEntity` in that it's not returning the `Entity` object but instead the XML Node of an Entity (the fact that you can pass in the type is not really that important just like it isn't for `getEntity`). Maybe `getEntityNode`, `getEntityTypeNode`?

Reply:
It's ok, that makes sense, leave it as you've got it.